### ISSUE TITLE
About label_data and unlabel_data

### ISSUE THREAD

**Issue by zjy359** [OatmealLiu/class-iNCD#4]
Hello,I want to kown when this net input label_data,what is the output?And unlabel_data?

**Comment by OatmealLiu** [comment]
Hi! Thanks for the question! Could you please further clarify your question? I don't get it. You can also quote the code or scripts to describe the question. Thanks.

Best regards,
Miu

**Comment by zjy359** [comment]
您好，我想问的是当网络训练完成后，也就是在测试阶段，输入无标签数据网络会输出什么呢？是伪标签还是其他的什么？输入有标签数据会输出具体的标签吗？

**Comment by OatmealLiu** [comment]
Hey! Thanks for the clarification of your question.

Before answering the question, one thing I wanna metion is that, In this work, we care both the classification performance for labeled data and the clustering performance for unlabeled data. In addition, unlike common clustering evaluation metrics, we also care about the positions (indices) of the clustering centers. Besides, during forward inference (test stage), our method can operate in a task-agnostic fashion.

Therefore, during forward inference (test stage), given an unseen sample `x` as input, our method will predict `the classification result (c-th label, if the logit for c-th has the highest probablity)` or `clustering result (k-th clustering center if the logit k-th clustering center has the highest probablity)`.

Let's use CIFAR-10 as an example:

During stage-1, we train our model in supervised fashion on the labeled data belongs to `[0, 8)` classes. Then, we train our model to discover novel classes in self-supervised fashion on the unlabeled data belongs to `[8, 10)` classes during stage-2. The novel class discovery (NCD) result of stage-2 are essentially `2 clustering centers with their fixed position in the logits` in this case.

So, to sum-up, after stage-1 and stage-2 trainings, our method can perform both classification task and task. In CIFAR-10 case, given an unseen sample `x` as input, our method will output either `the predicted label (c-th label, if the logit for c-th has the highest probablity, c belongs to [0, 8), in which we know the semantic label for each)` or `the predicted index of the clustering center (k-th clustering center if the logit k-th clustering center has the highest probablity, k belongs to [8, 10), in whcih we do not know the semantic label for each but we know they are novel classes)`.

Hope this answer can help you!

Best regards,
Miu

**Comment by zjy359** [comment]
Thanks for your reply,I understanding about my question.Thank you very much!!!


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "OatmealLiu/class-iNCD"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月29日(星期四) 下午5:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [OatmealLiu/class-iNCD] About label_data and unlabel_data (Issue #4)





 
Hey! Thanks for the clarification of your question.
 
Before answering the question, one thing I wanna metion is that, In this work, we care both the classification performance for labeled data and the clustering performance for unlabeled data. In addition, unlike common clustering evaluation metrics, we also care about the positions (indices) of the clustering centers. Besides, during forward inference (test stage), our method can operate in a task-agnostic fashion.
 
Therefore, during forward inference (test stage), given an unseen sample x as input, our method will predict the classification result (c-th label, if the logit for c-th has the highest probablity) or clustering (k-th clustering center if the logit k-th clustering center has the highest probablity) result.
 
Let's use CIFAR-10 as an example. During stage-1, we train our model in supervised fashion on the labeled data belongs to [0, 8) classes. Then, we train our model to discover novel classes in self-supervised fashion on the unlabeled data belongs to [8, 10) classes during stage-2. The novel class discovery (NCD) result of stage-2 is essentially 2 clustering centers with their fixed position in the logits in this case. So, to sum-up, after stage-1 and stage-2 trainings, our method can perform both classification task and task. In CIFAR-10 case, given an unseen sample x as input, our method will output either `the classified label (c-th label, if the logit for c-th has the highest probablity, C \in [0, 8))
